File: Arrays/1-two-sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """        
        index_map = {}  # key: number, value: index

        for idx, num in enumerate(nums):
            complement = target - num
            if complement in index_map:
                return [index_map[complement], idx]
            index_map[num] = idx

        raise ValueError("No solution exists")

File: Arrays/1-two-sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map(); // key: number, value: index

    for (let idx = 0; idx < nums.length; idx++) {
        const complement = target - nums[idx];

        if (map.has(complement)) {
            return [map.get(complement), idx];
        }

        map.set(nums[idx], idx);
    }
    throw new Error("No solution exists");
};